caproto/_log.py

```diff
@@ -143,10 +156,120 @@ def color_logs(color):
 
 
 logger = logging.getLogger('caproto')
+ch_logger = logging.getLogger('caproto.ch')
+search_logger = logging.getLogger('caproto.bcast.search')
 current_handler = None  # overwritten below
 
 
-def set_handler(file=sys.stdout, datefmt='%H:%M:%S', color=True):
+class LoggerNameFilter(logging.Filter):
+    def __init__(self, logger_names, level='NOSET'):
+        self.logger_names = logger_names
+        self.levelno = validate_level(level)
+    def filter(self, record):
+        for i in self.logger_names:
+            if fnmatch.fnmatch(record.name, i) and record.levelno >= self.levelno:
+                return True
+        return False
+
+
+def validate_level(level)-> int:
+    '''
+    Return a int for level comparison
+
+    '''
+    if isinstance(level, int):
+        levelno = level
+    elif isinstance(level, str):
+        levelno = logging.getLevelName(level)
+
+    if isinstance(levelno, int):
+        return levelno
+    else:
+        raise ValueError('argument level should be either one of string in LOG LEVEL or int')
```

Comment:
In caproto, any time we raise an exception, we raise a caproto-specific subclass so that downstream callers can tell that the error originated in caproto. This should be a `CaprotoValueError`, imported from `._utils`. Same for all `ValueError` instances below.